test/rb.c

```diff
@@ -73,50 +75,73 @@ check_simple(tree_t *tree)
 			fail_unless(test_prev(tree, node) == NULL);
 		}
 	}
-	fail_if(test_search(tree, NUMBER_NODES) != NULL);
-	fail_unless(test_first(tree)->key == 0);
-	fail_unless(test_last(tree)->key == NUMBER_NODES - 1);
+	ok(test_search(tree, NUMBER_NODES) == NULL);
+	ok(test_first(tree)->key == 0);
+	ok(test_last(tree)->key == NUMBER_NODES - 1);
+
 	footer();
+	check_plan();
 	return nodes;
 }
 
 static node_t *
-print_cb(tree_t *t, node_t *node, void* arg)
+check_forward_cb(tree_t *t, node_t *node, void *arg)
+{
+	(void)t;
+	int i = *(int *)arg;
+	fail_unless(node->key == i && node->data == 2 * i);
+	(*(int *)arg)++;
+	return NULL;
+}
+
+static node_t *
+check_reverse_cb(tree_t *t, node_t *node, void *arg)
 {
 	(void)t;
-	(void)arg;
-	printf(" %i->%i", node->key, node->data);
+	(*(int *)arg)--;
+	int i = *(int *)arg;
+	fail_unless(node->key == i && node->data == 2 * i);
```

Comment:
Why not `ok` here and in `check_forward_cb`?

Reply:
Changed to use `ok`.